### DISCUSSION TITLE
KKS Action?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#15801] [Q&A Modules: Phase field] [answered]
Would it be possible to put together an Action for a KKS model much like the existing Conserved/Nonconserved phase field actions? Seems like it should be straightforward enough for a two phase system at least.

I do see that a [KKSAction](https://github.com/idaholab/moose/blob/next/modules/phase_field/src/action/KKSAction.C) was started in the phase field module, but it's basically empty at the moment.

**Comment by jessecarterMOOSE** [comment, marked as answer]
ping @dschwen

**Reply by dschwen** [reply]
Pong!

Yeah, that would be a nice easy project for someone with time on their hands. Maybe one of our incoming new hires.

**Reply by dschwen** [reply]
It's quite amazing that this husk of an action was actually merged into our code base. I bet I had grand plans back then :-D

**Comment by aeslaughter** [comment]
@jessecarterMOOSE Are you able to mark as answered? I am not really sure who has the power?

**Comment by jessecarterMOOSE** [comment]
@dschwen  I asked because I do have time at the moment (thanks to all the working from home), and I could use it in my app. I guess I was trying to get a feel for whether it was possible or if there was some complication I wasn't seeing.

**Reply by dschwen** [reply]
Did you ever take a stab at this?

**Reply by jessecarterMOOSE** [reply]
Unfortunately no. I looked around the examples saw a few different variations, for instance when coupling in elastic energy, and just ended up confusing myself. I'm still interested provided I can wrap my head around it!

**Comment by anilkunwar** [comment]
@jessecarterMOOSE : it is nice to know that KKSAction was originally envisioned for addressing the volume conservation and non-conservation in the multi-phase field model (KKS method).

**Reply by jessecarterMOOSE** [reply]
Which makes it all the more daunting as a task!

**Comment by jessecarterMOOSE** [comment]
@dschwen Do you envision a KKS action as being a single action rather than Conserved/Nonconserved? And as far as I can tell, it doesn't need as many formulations (direct, forward split, reverse split), at least not in practice.

**Reply by dschwen** [reply]
single action is fine, as the conserved variables are intertwined with the nonconserved variables.
